kratos/tests/cpp_tests/utilities/test_nearest_point_utilities.cpp

```diff
@@ -20,8 +21,96 @@
 #include "geometries/triangle_3d_3.h"
 #include "utilities/geometry_utilities/nearest_point_utilities.h"
 
+// // Debugging
+// #include "input_output/vtk_output.h"
+
 namespace Kratos::Testing {
 
+// namespace
+// {
+//     // This method is for debugging
+//     void WriteVTK(ModelPart& rModelPart)
+//     {
+//         auto volume_vtk_output_parameters = Parameters(R"({
+//             "model_part_name"                    : "})"+rModelPart.Name()+R"({",
+//             "file_format"                        : "ascii",
+//             "output_precision"                   : 8,
+//             "output_path"                        : ")"+rModelPart.Name()+R"(",
+//             "nodal_data_value_variables"         : [],
+//             "nodal_flags"                        : [],
+//             "condition_data_value_variables"     : [],
+//             "condition_flags"                    : []
+//         })");
+//         VtkOutput volume_vtk_output(rModelPart, volume_vtk_output_parameters);
+//         volume_vtk_output.PrintOutput();
+//     }
+// }
+
+/**
+ * @brief Finds the nearest point to the given point on a triangle. 
+ * @details It first projects the point into the triangle surface. If the projected point is inside the triangle 
+ * it returns the projected point. If not it returns the nearest point on the edges of the triangle.
+ * Dividing the plane of the triangle in 7 zones and find the nearest reflecting those zones
+ *
+ *             \       /
+ *              \  6  /
+ *               \   /
+ *                \ /
+ *                 /\
+ *                /  \
+ *          2    /    \     3
+ *              /   1  \
+ *             /        \
+ *    _______ /__________\_____________
+ *           /            \
+ *     5    /       4      \    7
+ *         /                \
+ *        /                  \
+ * @tparam Type of the Point
+ * @tparam TGeometryType The type of the triangle. Assumes to have [] access and IsInside method
+ * @param rPoint The query point which we want to get nearest point to it on the line
+ * @param rTriangle The triangle in which we want to find the nearest point to rPoint
+ * @return The nearest point to rPoint
+ */
+template<class TPointType, class TGeometryType>
```

Comment:
I'd put this into an anonymous namespace so we completely avoid using it outside the context of this test.